experimental/casemap/src/closer.rs

```diff
@@ -0,0 +1,217 @@
+    #[cfg(skip)]
+    functions: [
+        try_new_from_casemapper_unstable,
+        try_new_with_mapper_with_any_provider,
+        try_new_with_mapper_with_buffer_provider,
+        try_new_with_mapper_unstable,
+        Self,
+    ]);
+
+    /// Construct this object to wrap an existing CaseMapper (or a reference to one), loading additional data as needed.
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    ///
+    /// <div class="stab unstable">⚠️ The bounds on <tt>provider</tt> may change over time, including in SemVer minor releases.</div>
+    pub fn try_new_with_mapper_unstable<P>(provider: &P, casemapper: CM) -> Result<Self, DataError>
+    where
+        P: DataProvider<CaseMapV1Marker> + DataProvider<CaseMapUnfoldV1Marker> + ?Sized,
+    {
+        let unfold = provider.load(Default::default())?.take_payload()?;
+        Ok(Self {
+            cm: casemapper,
+            unfold,
+        })
+    }
+
+    /// Adds all simple case mappings and the full case folding for `c` to `set`.
+    /// Also adds special case closure mappings.
+    ///
+    /// In other words, this adds all strings/characters that this casemaps to, as
+    /// well as all characters that may casemap to this one.
+    ///
+    /// The character itself is not added.
+    ///
+    /// For example, the mappings
+    /// - for s include long s
+    /// - for sharp s include ss
+    /// - for k include the Kelvin sign
+    ///
+    /// This function is identical to [`CaseMapper::add_case_closure()`]; if you don't
+    /// need [`Self::add_string_case_closure()`] consider using a [`CaseMapper`] to avoid
+    /// loading additional data.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use icu_casemap::CaseMapCloser;
+    /// use icu_collections::codepointinvlist::CodePointInversionListBuilder;
+    ///
+    /// let cm = CaseMapCloser::new();
+    /// let mut builder = CodePointInversionListBuilder::new();
+    /// cm.add_case_closure('s', &mut builder);
+    ///
+    /// let set = builder.build();
+    ///
+    /// assert!(set.contains('S'));
+    /// assert!(set.contains('ſ'));
+    /// assert!(!set.contains('s')); // does not contain itself
+    /// ```
+    pub fn add_case_closure<S: ClosureSet>(&self, c: char, set: &mut S) {
+        self.cm.as_ref().add_case_closure(c, set);
+    }
+
+    /// Finds all characters and strings which may casemap to `s` and adds them to the set.
+    ///
+    /// The string itself is not added to the set.
+    ///
+    /// Returns true if the string was found
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use icu_casemap::CaseMapCloser;
+    /// use icu_collections::codepointinvlist::CodePointInversionListBuilder;
+    ///
+    /// let cm = CaseMapCloser::new();
+    /// let mut builder = CodePointInversionListBuilder::new();
+    /// let found = cm.add_string_case_closure("ffi", &mut builder);
+    /// assert!(found);
+    /// let set = builder.build();
```

Comment:
Question: what happens if my string is `ffiss`? Do I get `ﬃss`, `ﬃß`, and `ffiß` all added to my set?

Reply:
nothing, it's only for the atomic casemappings.

I don't think there's good terminology around this unfortunately.